shotgun_api3/shotgun.py

```diff
@@ -3810,19 +3809,19 @@ def _upload_data_to_storage(self, data, content_type, size, storage_url):
         :rtype: str
         """
         try:
-            opener = urllib2.build_opener(urllib2.HTTPHandler)
+            opener = url_request.build_opener(url_request.HTTPHandler)
 
-            request = urllib2.Request(storage_url, data=data)
+            request = url_request.Request(storage_url, data=data)
             request.add_header("Content-Type", content_type)
             request.add_header("Content-Length", size)
             request.get_method = lambda: "PUT"
             result = opener.open(request)
             etag = result.info().getheader("ETag")
```

Comment:
```suggestion
             etag = result.headers["ETag"]
```
Changing this got it working for me (in 3.7.2), but there's probably a more robust solution.